File: app/src/main/java/ru/samtakoy/core/business/CoursesPlanner.java
package ru.samtakoy.core.business;

import androidx.annotation.Nullable;

import java.util.List;

import ru.samtakoy.core.database.room.entities.elements.Schedule;

public interface CoursesPlanner {


    // проверить позднее - есть ли начатые незавершенные обучения/повторения и показать нотификацию
    void planUncompletedTasksChecking();

    void reScheduleLearnCourses();

    void planAdditionalCards(Long qPackId, List<Long> errorCardIds, Schedule schedule);

    void planAdditionalCards(Long qPackId, @Nullable String subTitle, List<Long> cardIds, Schedule restSchedule);
}

File: module/features/learncourse/api/src/commonMain/kotlin/ru/samtakoy/domain/learncourse/CoursesPlanner.kt
package ru.samtakoy.domain.learncourse

import ru.samtakoy.domain.learncourse.schedule.Schedule

interface CoursesPlanner {
    // проверить позднее - есть ли начатые незавершенные обучения/повторения и показать нотификацию
    fun planUncompletedTasksChecking()

    fun reScheduleLearnCourses()

    fun planAdditionalCards(qPackId: Long, errorCardIds: List<Long>, schedule: Schedule)

    // todo suspend
    fun planAdditionalCards(qPackId: Long, subTitle: String?, cardIds: List<Long>, restSchedule: Schedule)
}
